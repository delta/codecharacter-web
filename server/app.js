const express = require('express');
const path = require('path');
const app = express();

app.use(express.static(path.join(__dirname, '..', 'build')));

app.get('/*', (req, res) => {
  res.sendFile(path.join(__dirname, '..', 'build', 'index.html'));
});

app.listen(5000, '0.0.0.0');
console.log('Server is listening on http://localhost:5000');
