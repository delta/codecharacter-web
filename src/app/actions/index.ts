export * from 'app/actions/Dashboard';
export * from 'app/actions/Leaderboard';
export * from 'app/actions/code/EditorSettings';
export * from 'app/actions/GameLog';
export * from 'app/actions/code/Code';
export * from 'app/actions/User';
export * from 'app/actions/Notification';
export * from 'app/actions/code/Submission';
export * from 'app/actions/MatchView';
export * from 'app/actions/ProfileUser';
