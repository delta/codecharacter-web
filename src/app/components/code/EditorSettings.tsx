import * as styles from 'app/styles/EditorSettings.module.css';
import { keyboardHandlers, themes } from 'app/types/code/Editor';
import * as EditorSettingsInterfaces from 'app/types/code/EditorSettings';
import classnames from 'classnames';
import * as React from 'react';
import { Col, FormGroup, Grid, Row } from 'react-bootstrap';

export class EditorSettings extends React.Component<EditorSettingsInterfaces.Props, {}> {
  public render() {
    const {
      fontSize,
      changeFontSize,
      theme,
      changeTheme,
      changeKeyboardHandler,
      enableBasicAutoCompletion,
      toggleBasicAutoCompletion,
      enableSnippets,
      keyboardHandler,
      toggleSnippets,
    } = this.props;

    const fontSizeOptions = [];
    for (let i = 8; i <= 40; i += 2) {
      fontSizeOptions.push(i);
    }

    return (
      <Grid fluid={true} className={classnames(styles.EditorSettings)}>
        <Row className="justify-content-between py-2 pl-3">
          <Col className="text-light font-weight-bold my-auto">SETTINGS</Col>
        </Row>
        <Row>
          <Col sm={12} className={classnames('mb-1', styles.setting)}>
            <div className="font-weight-bold py-1">Font Size</div>
            <FormGroup controlId="fontSize">
              <select
                className={classnames(
                  'form-control',
                  styles.formControlSelect,
                  'font-size-control',
                )}
                value={fontSize}
                onChange={(e) => changeFontSize(Number(e.target.value))}
              >
                {fontSizeOptions.map((font: number) => (
                  <option value={font} key={font}>
                    {font}
                  </option>
                ))}
              </select>
            </FormGroup>
          </Col>
          <Col sm={12} className={classnames('mb-1', styles.setting)}>
            <div className="font-weight-bold py-1">Editor Theme</div>
            <FormGroup controlId="theme">
              <select
                className={classnames('form-control', styles.formControlSelect, 'theme-control')}
                value={theme}
                onChange={(e) => changeTheme(e.target.value)}
              >
                {themes.map((themeValue: string) => (
                  <option value={themeValue} key={themeValue}>
                    {themeValue}
                  </option>
                ))}
              </select>
            </FormGroup>
          </Col>
          <Col sm={12} className={classnames('mb-1', styles.setting)}>
            <div className="font-weight-bold py-1">Editor Keybinding</div>
            <FormGroup controlId="keyboardHandler">
              <select
                className={classnames(
                  'form-control',
                  styles.formControlSelect,
                  'keyboardHandler-control',
                )}
                value={keyboardHandler}
                onChange={(e) => changeKeyboardHandler(e.target.value)}
              >
                {keyboardHandlers.map((keyboardHandlerOption: string) => (
                  <option value={keyboardHandlerOption} key={keyboardHandlerOption}>
                    {keyboardHandlerOption}
                  </option>
                ))}
              </select>
            </FormGroup>
          </Col>
          <Col sm={12} className={classnames('mb-1', styles.setting)}>
            <div className="font-weight-bold py-1">Basic Auto Completion</div>
            <FormGroup controlId="autoCompletion">
              <select
                className={classnames(
                  'form-control',
                  styles.formControlSelect,
                  'auto-complete-control',
                )}
                value={enableBasicAutoCompletion ? 'enabled' : 'disabled'}
                onChange={(e) =>
                  toggleBasicAutoCompletion(e.target.value === 'enabled' ? true : false)
                }
              >
                <option value={'enabled'} key={'enabled'}>
                  Enabled
                </option>
                <option value={'disabled'} key={'disabled'}>
                  Disabled
                </option>
              </select>
            </FormGroup>
          </Col>
          <Col sm={12} className={classnames('mb-1', styles.setting)}>
            <div className="font-weight-bold py-1">Snippets</div>
            <FormGroup controlId="snippets">
              <select
                className={classnames('form-control', styles.formControlSelect, 'snippets-control')}
                value={enableSnippets ? 'enabled' : 'disabled'}
                onChange={(e) => toggleSnippets(e.target.value === 'enabled' ? true : false)}
              >
                <option value={'enabled'} key={'enabled'}>
                  Enabled
                </option>
                <option value={'disabled'} key={'disabled'}>
                  Disabled
                </option>
              </select>
            </FormGroup>
          </Col>
        </Row>
      </Grid>
    );
  }
}
