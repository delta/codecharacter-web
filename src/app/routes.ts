export enum Routes {
  ROOT = '/',
  LOGIN = '/login',
  REGISTER = '/register',
  LEADERBOARD = '/leaderboard',
  USER_PROFILE_MODEL = '/profile',
  GITHUB_OAUTH = '/login/github',
  GOOGLE_OAUTH = '/login/google',
  USER_ACTIVATION = '/user-activate',
}
