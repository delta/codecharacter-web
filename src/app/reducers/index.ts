import { codeReducer } from 'app/reducers/code/Code';
import { editorReducer } from 'app/reducers/code/Editor';
import { submissionReducer } from 'app/reducers/code/Submission';
import { dashboardReducer } from 'app/reducers/Dashboard';
import { gameLogReducer } from 'app/reducers/GameLog';
import { leaderboardReducer } from 'app/reducers/Leaderboard';
import { matchesReducer } from 'app/reducers/MatchView';
import { notificationReducer } from 'app/reducers/Notification';
import { userReducer } from 'app/reducers/User';
import * as CodeInterfaces from 'app/types/code/Code';
import * as EditorInterfaces from 'app/types/code/Editor';
import * as SubmissionInterfaces from 'app/types/code/Submission';
import * as DashboardInterfaces from 'app/types/Dashboard';
import * as GameLogInterfaces from 'app/types/GameLog';
import * as LeaderboardInterfaces from 'app/types/Leaderboard';
import * as MatchInterfaces from 'app/types/MatchView';
import * as NotificationInterfaces from 'app/types/Notification';
import * as UserInterfaces from 'app/types/User';
import { routerReducer, RouterState } from 'react-router-redux';
import { combineReducers } from 'redux';

export const rootReducer = combineReducers({
  code: codeReducer,
  dashboard: dashboardReducer,
  editor: editorReducer,
  gameLog: gameLogReducer,
  leaderboard: leaderboardReducer,
  match: matchesReducer,
  notification: notificationReducer,
  router: routerReducer,
  submission: submissionReducer,
  user: userReducer,
});

export interface RootState {
  code: CodeInterfaces.CodeStoreState;
  dashboard: DashboardInterfaces.DashboardStoreState;
  editor: EditorInterfaces.EditorStoreState;
  leaderboard: LeaderboardInterfaces.LeaderboardStoreState;
  match: MatchInterfaces.MatchStoreState;
  notification: NotificationInterfaces.NotificationStoreState;
  router: RouterState;
  gameLog: GameLogInterfaces.GameLogStoreState;
  user: UserInterfaces.UserStoreState;
  submission: SubmissionInterfaces.SubmissionStoreState;
}
