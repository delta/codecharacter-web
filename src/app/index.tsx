import ActivateUser from 'app/containers/Authentication/ActivateUser';
import Login from 'app/containers/Authentication/Login';
import Register from 'app/containers/Authentication/Register';
import Dashboard from 'app/containers/Dashboard';
import LandingPage from 'app/containers/LandingPage';
import Leaderboard from 'app/containers/Leaderboard';
import UserProfileModal from 'app/containers/UserProfileModal';
import { Routes } from 'app/routes';
// @ts-ignore
import { initializeRendererAssets } from 'code-character-renderer-20';
import * as React from 'react';
import { hot } from 'react-hot-loader';
// @ts-ignore
import { BrowserRouter, Route, Switch } from 'react-router-dom';

initializeRendererAssets();

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <BrowserRouter>
    <Switch>
      <Route exact path={Routes.ROOT} component={Dashboard} />
      <Route exact path={Routes.LOGIN} component={Login} />
      <Route exact path={Routes.REGISTER} component={Register} />
      <Route exact path={Routes.LEADERBOARD} component={Leaderboard} />
      <Route exact path={Routes.USER_PROFILE_MODEL} component={UserProfileModal} />
      <Route path={Routes.USER_ACTIVATION} component={ActivateUser} />
      <Route exact path={Routes.HOME} component={LandingPage} />
    </Switch>
  </BrowserRouter>
));
