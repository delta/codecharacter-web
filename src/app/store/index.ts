import { logger } from 'app/middleware';
import { rootReducer } from 'app/reducers';
import { codeSagas } from 'app/sagas/Code';
import { leaderboardSagas } from 'app/sagas/Leaderboard';
import { matchSagas } from 'app/sagas/MatchView';
import { notificationSagas } from 'app/sagas/Notification';
import { submissionSagas } from 'app/sagas/Submission';
import { userSagas } from 'app/sagas/User';
import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/es/storage';
// tslint:disable-next-line:import-name
import createSagaMiddleware from 'redux-saga';

export function configureStore(initialState?: object) {
  const sagaMiddleware = createSagaMiddleware();
  let middleware = applyMiddleware(logger, sagaMiddleware);

  if (process.env.NODE_ENV !== 'production') {
    middleware = composeWithDevTools(middleware);
  }

  const persistConfig = {
    storage,
    blacklist: ['gameLog', 'submission'],
    key: 'root',
  };

  const persistedReducer = persistReducer(persistConfig, rootReducer);

  const store = createStore(persistedReducer, initialState, middleware);
  sagaMiddleware.run(userSagas);
  sagaMiddleware.run(codeSagas);
  sagaMiddleware.run(leaderboardSagas);
  sagaMiddleware.run(submissionSagas);
  sagaMiddleware.run(matchSagas);
  sagaMiddleware.run(notificationSagas);

  const persistor = persistStore(store);

  if (module.hot) {
    module.hot.accept('app/reducers', () => {
      const nextReducer = require('app/reducers');
      store.replaceReducer(nextReducer);
    });
  }

  return { store, persistor };
}
