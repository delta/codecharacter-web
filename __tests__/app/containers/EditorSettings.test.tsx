import EditorSettings from 'app/containers/code/EditorSettings';
import { configureStore } from 'app/store';
import { mount } from 'enzyme';
import * as React from 'react';

describe('EditorSettings Container', () => {
  const { store } = configureStore();

  it('Should render EditorSettings', () => {
    const wrapper = mount(<EditorSettings />, {
      context: {
        store,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Should change font', () => {
    const wrapper = mount(<EditorSettings />, {
      context: {
        store,
      },
    });

    expect(wrapper.find('select.font-size-control').props().value).toBe(16);
    wrapper.find('select.font-size-control').simulate('change', { target: { value: '32' } });
    expect(wrapper.find('select.font-size-control').props().value).toBe(32);
  });

  it('Should change theme', () => {
    const wrapper = mount(<EditorSettings />, {
      context: {
        store,
      },
    });

    expect(wrapper.find('select.theme-control').props().value).toBe('monokai');
    wrapper.find('select.theme-control').simulate('change', { target: { value: 'github' } });
    expect(wrapper.find('select.theme-control').props().value).toBe('github');
  });

  it('Should enable autoComplete', () => {
    const wrapper = mount(<EditorSettings />, {
      context: {
        store,
      },
    });

    expect(wrapper.find('select.auto-complete-control').props().value).toBe('enabled');
    wrapper
      .find('select.auto-complete-control')
      .simulate('change', { target: { value: 'disabled' } });
    expect(wrapper.find('select.auto-complete-control').props().value).toBe('disabled');
  });

  it('Should enable snippets', () => {
    const wrapper = mount(<EditorSettings />, {
      context: {
        store,
      },
    });

    expect(wrapper.find('select.snippets-control').props().value).toBe('enabled');
    wrapper.find('select.snippets-control').simulate('change', { target: { value: 'disabled' } });
    expect(wrapper.find('select.snippets-control').props().value).toBe('disabled');
  });
});
