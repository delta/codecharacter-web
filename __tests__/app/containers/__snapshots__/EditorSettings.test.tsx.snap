// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorSettings Container Should render EditorSettings 1`] = `
<Connect(EditorSettings)>
  <EditorSettings
    changeFontSize={[Function]}
    changeKeyboardHandler={[Function]}
    changeTheme={[Function]}
    enableBasicAutoCompletion={true}
    enableSnippets={true}
    fontSize={16}
    keyboardHandler="default"
    theme="solarized_light"
    toggleBasicAutoCompletion={[Function]}
    toggleSnippets={[Function]}
  >
    <Grid
      bsClass="container"
      className=""
      componentClass="div"
      fluid={true}
    >
      <div
        className="container-fluid"
      >
        <Row
          bsClass="row"
          className="justify-content-between py-2 pl-3"
          componentClass="div"
        >
          <div
            className="justify-content-between py-2 pl-3 row"
          >
            <Col
              bsClass="col"
              className="text-light font-weight-bold my-auto"
              componentClass="div"
            >
              <div
                className="text-light font-weight-bold my-auto"
              >
                SETTINGS
              </div>
            </Col>
          </div>
        </Row>
        <Row
          bsClass="row"
          componentClass="div"
        >
          <div
            className="row"
          >
            <Col
              bsClass="col"
              className="mb-1"
              componentClass="div"
              sm={12}
            >
              <div
                className="mb-1 col-sm-12"
              >
                <div
                  className="font-weight-bold py-1"
                >
                  Font Size
                </div>
                <FormGroup
                  bsClass="form-group"
                  controlId="fontSize"
                >
                  <div
                    className="form-group"
                  >
                    <select
                      className="form-control font-size-control"
                      onChange={[Function]}
                      value={16}
                    >
                      <option
                        key="8"
                        value={8}
                      >
                        8
                      </option>
                      <option
                        key="10"
                        value={10}
                      >
                        10
                      </option>
                      <option
                        key="12"
                        value={12}
                      >
                        12
                      </option>
                      <option
                        key="14"
                        value={14}
                      >
                        14
                      </option>
                      <option
                        key="16"
                        value={16}
                      >
                        16
                      </option>
                      <option
                        key="18"
                        value={18}
                      >
                        18
                      </option>
                      <option
                        key="20"
                        value={20}
                      >
                        20
                      </option>
                      <option
                        key="22"
                        value={22}
                      >
                        22
                      </option>
                      <option
                        key="24"
                        value={24}
                      >
                        24
                      </option>
                      <option
                        key="26"
                        value={26}
                      >
                        26
                      </option>
                      <option
                        key="28"
                        value={28}
                      >
                        28
                      </option>
                      <option
                        key="30"
                        value={30}
                      >
                        30
                      </option>
                      <option
                        key="32"
                        value={32}
                      >
                        32
                      </option>
                      <option
                        key="34"
                        value={34}
                      >
                        34
                      </option>
                      <option
                        key="36"
                        value={36}
                      >
                        36
                      </option>
                      <option
                        key="38"
                        value={38}
                      >
                        38
                      </option>
                      <option
                        key="40"
                        value={40}
                      >
                        40
                      </option>
                    </select>
                  </div>
                </FormGroup>
              </div>
            </Col>
            <Col
              bsClass="col"
              className="mb-1"
              componentClass="div"
              sm={12}
            >
              <div
                className="mb-1 col-sm-12"
              >
                <div
                  className="font-weight-bold py-1"
                >
                  Editor Theme
                </div>
                <FormGroup
                  bsClass="form-group"
                  controlId="theme"
                >
                  <div
                    className="form-group"
                  >
                    <select
                      className="form-control theme-control"
                      onChange={[Function]}
                      value="solarized_light"
                    >
                      <option
                        key="monokai"
                        value="monokai"
                      >
                        monokai
                      </option>
                      <option
                        key="github"
                        value="github"
                      >
                        github
                      </option>
                      <option
                        key="tomorrow"
                        value="tomorrow"
                      >
                        tomorrow
                      </option>
                      <option
                        key="kuroir"
                        value="kuroir"
                      >
                        kuroir
                      </option>
                      <option
                        key="twilight"
                        value="twilight"
                      >
                        twilight
                      </option>
                      <option
                        key="xcode"
                        value="xcode"
                      >
                        xcode
                      </option>
                      <option
                        key="textmate"
                        value="textmate"
                      >
                        textmate
                      </option>
                      <option
                        key="solarized_dark"
                        value="solarized_dark"
                      >
                        solarized_dark
                      </option>
                      <option
                        key="solarized_light"
                        value="solarized_light"
                      >
                        solarized_light
                      </option>
                      <option
                        key="terminal"
                        value="terminal"
                      >
                        terminal
                      </option>
                      <option
                        key="chaos"
                        value="chaos"
                      >
                        chaos
                      </option>
                      <option
                        key="chrome"
                        value="chrome"
                      >
                        chrome
                      </option>
                      <option
                        key="eclipse"
                        value="eclipse"
                      >
                        eclipse
                      </option>
                      <option
                        key="dracula"
                        value="dracula"
                      >
                        dracula
                      </option>
                    </select>
                  </div>
                </FormGroup>
              </div>
            </Col>
            <Col
              bsClass="col"
              className="mb-1"
              componentClass="div"
              sm={12}
            >
              <div
                className="mb-1 col-sm-12"
              >
                <div
                  className="font-weight-bold py-1"
                >
                  Editor Keybinding
                </div>
                <FormGroup
                  bsClass="form-group"
                  controlId="keyboardHandler"
                >
                  <div
                    className="form-group"
                  >
                    <select
                      className="form-control keyboardHandler-control"
                      onChange={[Function]}
                      value="default"
                    >
                      <option
                        key="emacs"
                        value="emacs"
                      >
                        emacs
                      </option>
                      <option
                        key="vim"
                        value="vim"
                      >
                        vim
                      </option>
                      <option
                        key="default"
                        value="default"
                      >
                        default
                      </option>
                    </select>
                  </div>
                </FormGroup>
              </div>
            </Col>
            <Col
              bsClass="col"
              className="mb-1"
              componentClass="div"
              sm={12}
            >
              <div
                className="mb-1 col-sm-12"
              >
                <div
                  className="font-weight-bold py-1"
                >
                  Basic Auto Completion
                </div>
                <FormGroup
                  bsClass="form-group"
                  controlId="autoCompletion"
                >
                  <div
                    className="form-group"
                  >
                    <select
                      className="form-control auto-complete-control"
                      onChange={[Function]}
                      value="enabled"
                    >
                      <option
                        key="enabled"
                        value="enabled"
                      >
                        Enabled
                      </option>
                      <option
                        key="disabled"
                        value="disabled"
                      >
                        Disabled
                      </option>
                    </select>
                  </div>
                </FormGroup>
              </div>
            </Col>
            <Col
              bsClass="col"
              className="mb-1"
              componentClass="div"
              sm={12}
            >
              <div
                className="mb-1 col-sm-12"
              >
                <div
                  className="font-weight-bold py-1"
                >
                  Snippets
                </div>
                <FormGroup
                  bsClass="form-group"
                  controlId="snippets"
                >
                  <div
                    className="form-group"
                  >
                    <select
                      className="form-control snippets-control"
                      onChange={[Function]}
                      value="enabled"
                    >
                      <option
                        key="enabled"
                        value="enabled"
                      >
                        Enabled
                      </option>
                      <option
                        key="disabled"
                        value="disabled"
                      >
                        Disabled
                      </option>
                    </select>
                  </div>
                </FormGroup>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Grid>
  </EditorSettings>
</Connect(EditorSettings)>
`;
